packages/venia-concept/src/components/ProductImageCarousel/carousel.js

```diff
@@ -1,136 +1,112 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import memoize from 'memoize-one';
+import React, { useCallback, useMemo } from 'react';
+import { arrayOf, bool, number, shape, string } from 'prop-types';
+import { useCarousel } from '@magento/peregrine';
 
 import { resourceUrl } from 'src/drivers';
 import Icon from 'src/components/Icon';
 import ChevronLeftIcon from 'react-feather/dist/icons/chevron-left';
 import ChevronRightIcon from 'react-feather/dist/icons/chevron-right';
-import classify from 'src/classify';
-import ThumbnailList from './thumbnailList';
+import { mergeClasses } from 'src/classify';
+import Thumbnail from './thumbnail';
 import defaultClasses from './carousel.css';
 import { transparentPlaceholder } from 'src/shared/images';
+import Image from 'src/components/Image';
 
 const ChevronIcons = {
     left: ChevronLeftIcon,
     right: ChevronRightIcon
 };
 
-class Carousel extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string,
-            currentImage: PropTypes.string,
-            imageContainer: PropTypes.string,
-            'chevron-left': PropTypes.string,
-            'chevron-right': PropTypes.string
-        }),
-        images: PropTypes.arrayOf(
-            PropTypes.shape({
-                label: PropTypes.string,
-                position: PropTypes.number,
-                disabled: PropTypes.bool,
-                file: PropTypes.string.isRequired
-            })
-        ).isRequired
-    };
+const DEFAULT_IMAGE_WIDTH = 640;
 
-    state = {
-        activeItemIndex: 0,
-        currentImageLoaded: false
-    };
-
-    updateActiveItemIndex = index => {
-        this.setState({ activeItemIndex: index });
-    };
-
-    // The spec does not guarantee a position parameter,
-    // so the rule will be to order items without position last.
-    // See https://github.com/magento/graphql-ce/issues/113.
-    // Memoize this expensive operation based on reference equality
-    // of the `images` array. Apollo cache should return a new array
-    // only when it does a new fetch.
-    sortAndFilterImages = memoize(items =>
-        items
-            .filter(i => !i.disabled)
-            .sort((a, b) => {
-                const aPos = isNaN(a.position) ? 9999 : a.position;
-                const bPos = isNaN(b.position) ? 9999 : b.position;
-                return aPos - bPos;
-            })
+const Chevron = ({ className, direction, onClick }) => {
+    return (
```

Comment:
In general, I'd like to see one component definition per module. I can see how this would be an exception since it's pretty thin and doesn't get all the usual ceremony...but I'd consider making it its own module.

Also, we shouldn't call this `Chevron` or use `chevron` in the classnames. Semantically, these are `previous` and `next` buttons, and they simply happen to be presented as chevrons at present, which is a detail that the component would ideally not know.

Reply:
I ended up removing the component entirely and am now re-using the generic `Button` component.